### ISSUE TITLE
Build error with --disable-wallet

### ISSUE THREAD

**Issue by ghost** [dashpay/dash#1357]
Error
```
rpcmining.cpp: In function 'UniValue getblocktemplate(const UniValue&, bool)':
rpcmining.cpp:466:10: error: 'masternodeSync' was not declared in this scope
     if (!masternodeSync.IsSynced())
```

Reproduce:
```
sudo docker run -it --rm alpine /bin/ash
```

```
apk update
apk add ca-certificates openssl wget git
apk add gcc g++ make autoconf automake libtool zeromq
apk add db-dev boost-dev zeromq-dev libevent-dev libressl-dev miniupnpc-dev
```

```
git clone https://github.com/dashpay/dash.git
cd dash
git checkout v0.12.1.1

./autogen.sh
./configure --disable-wallet
make
```

When building with `./configure --with-incompatible-bdb` the build ends fine and the blockchain syncs without any trouble.  Same happens in v0.12.1.0, no v0.12.0.x tested because they do not support libre ssl.

**Comment by thelazier** [comment]
The error should be solved by #1358 .
`--with-incompatible-bdb` is not related to supporting libre ssl , IMO. It's about db-dev in alpine package is not in version 2.8 which is required to compatible with official released which means the db files managed by this compiled binaries will not be able to used with officially released binaries.

**Comment by Nitya-Sattva** [comment]
Thx for the fix, seems to build.

I'm aware this is not related to ssl, mentioned here because I was to lazy to setup OpenSSL in my build setup to test version before 0.12.1.0. (I see, mentioned with a bad edit; changed to "tested because")

**Comment by thelazier** [comment]
Yep, the fix is not completed to make --disable-wallet works. I'm sorry.

**Comment by agulab** [comment]
Same problem here. I'm trying to cross compile for aarch64 with --disable-wallet on v0.12.1.3 and found the same error:
```
rpcmining.cpp: In function ‘UniValue getblocktemplate(const UniValue&, bool)’:
rpcmining.cpp:466:10: error: ‘masternodeSync’ was not declared in this scope
     if (!masternodeSync.IsSynced())
```

**Comment by Nitya-Sattva** [comment]
@agulab Building with "--with-incompatible-bdb" works fine on Arch for me, and then you can disable the wallet in the configuration file. I don't think this issue has an easy fix atm.

Building with the correct DB as a static library is not that hard either. Instructions in the doc's worked fine for me on arch and alpine. Look for the "Berkeley DB" section here: https://github.com/dashpay/dash/blob/master/doc/build-unix.md

**Comment by bitkevin** [comment]
I am using release version: `https://github.com/dashpay/dash/archive/v0.12.1.5.tar.gz`

```
./configure --disable-wallet
make
```

will get error:

```
In file included from ./wallet/walletdb.h:10:0,
                 from wallet/wallet.h:20,
                 from darksend.h:9,
                 from governance-vote.cpp:5:
./wallet/db.h:21:20: fatal error: db_cxx.h: No such file or directory
 #include <db_cxx.h>
                    ^
compilation terminated.
In file included from ./wallet/walletdb.h:10:0,
                 from wallet/wallet.h:20,
                 from darksend.h:9,
                 from governance-object.cpp:6:
./wallet/db.h:21:20: fatal error: db_cxx.h: No such file or directory
 #include <db_cxx.h>
                    ^
compilation terminated.
In file included from ./wallet/walletdb.h:10:0,
                 from wallet/wallet.h:20,
                 from darksend.h:9,
                 from main.cpp:41:
./wallet/db.h:21:20: fatal error: db_cxx.h: No such file or directory
 #include <db_cxx.h>
                    ^
compilation terminated.
In file included from ./wallet/walletdb.h:10:0,
                 from wallet/wallet.h:20,
                 from activemasternode.h:10,
                 from init.cpp:46:
./wallet/db.h:21:20: fatal error: db_cxx.h: No such file or directory
 #include <db_cxx.h>
                    ^
compilation terminated.
In file included from ./wallet/walletdb.h:10:0,
                 from wallet/wallet.h:20,
                 from darksend.h:9,
                 from governance.cpp:5:
./wallet/db.h:21:20: fatal error: db_cxx.h: No such file or directory
 #include <db_cxx.h>
                    ^
compilation terminated.
In file included from ./wallet/walletdb.h:10:0,
                 from wallet/wallet.h:20,
                 from net.cpp:22:
./wallet/db.h:21:20: fatal error: db_cxx.h: No such file or directory
 #include <db_cxx.h>
                    ^
```

**Comment by UdjinM6** [comment]
Should be fixed by #1733.
